std/haxe/macro/Compiler.hx

```diff
@@ -416,6 +416,17 @@ class Compiler {
 		#end
 	}
 
+	#if eval
+	/**
+		Enable null safety for the specified path.
+		E.g. `--macro nullSafety('my.pack')` to enable it for all the types in `my.pack` and in sub packages.
+		Or `--macro nullSafety('my.pack.MyClass') to enable it for `my.pack.MyClass` only
+	**/
+	public static function nullSafety(dotPath:String):Void {
```

Comment:
Somebody is probably going to complain about a missing `recursive` parameter for this.

I wonder if we should utilize `addGlobalMetadata` here here like we do for `keep`. That way you could just add `@:checkNull` or something and then process the types that have this metadata.

Reply:
Indeed meta sounds more consistent with what we already have in the compiler.
I'll do it `@:safety(true)` to enable null safety and `@:safety(false)` to exclude types/fields from null safety.

Reply:
Hm, to me it sounded like Simn merely suggested to use `addGlobalMetadata()` within `nullSafety()`, rather than remove it entirely?

Reply:
Indeed